modules/internal/ChapelArray.chpl

```diff
@@ -1815,17 +1823,44 @@ module ChapelArray {
             return true;
           }
         }
+        // We didn't find it, so return false.
+        return false;
+
+      // workaround for #21749:
+      } else if __primitive("call and fn resolves", "max", fullIdxType) {
+        var foundIt = false;
+        var locIdx = max(fullIdxType);
+        forall i in this.domain with (min reduce locIdx, max reduce foundIt) {
+          if this[i] == val {
+            locIdx = i;
+            foundIt = true;
```

Comment:
While musing about this—and general iterators—over the weekend, I realized that this will be buggy in the presence of multiple matches within a single task since each update replaces the index regardless of its value—should be guarded by a conditional.  I'll get on this tomorrow.